__tests__/app.test.js

```diff
@@ -0,0 +1,81 @@
+const request = require('supertest');
+const app = require('../app');
+const db = require('../db/connection');
+const seed = require('../db/seeds/seed');
+const testData = require('../db/data/test-data/index');
+
+beforeEach(() => {
+	return seed(testData);
+});
+
+afterAll(() => {
+	return db.end();
+});
+
+describe('Topics', () => {
+    
+    test('GET /api/topics should respond with a 200 status code', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200);
+    });
+    
+    test('GET /api/topics response should contain an array of topics', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then((response) => {
+            expect(response.body).toHaveProperty('topics');
+            expect(response.body.topics).toBeInstanceOf(Array);
+        });
+    });
+
+    test('GET /api/topics response should be the correct length (same as topicData length)', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then((response) => {
+            const topics = response.body.topics;
+            expect(topics).toHaveLength(testData.topicData.length);
+        });
+    });
+
+    test('GET /api/topics topics should have properties: description & slug', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then((response) => {
+            const topics = response.body.topics;
+            topics.forEach((topic) => {
+                expect(topic).toMatchObject({
+                    description: expect.any(String),
+                    slug: expect.any(String)
+                });
+            }); // foreach
+        }); // response
+    }); // test
+
+});
```

Comment:
I like this thoroughness but you could and for some reasons should have this all within the one test. I would assert the length of the array as an actual number for robustness (instead of taking the length from the data) and more importantly we should include this length assertion with the test that has the forEach. If for any reason an empty array/not an array is sent back then that test will not reach the topics.forEach so will give a false positive. The expected length assertion acts as a safety check as well as a check in its own right.